lib/src/main/kotlin/starknet/provider/service/HttpService.kt

```diff
@@ -34,15 +37,13 @@ class HttpService(private val url: String, private val method: String) : Service
             if (responseBody != null) {
                 return Json.decodeFromString(deserializer, responseBody.toString())
             } else {
-                // TODO: Create custom exception
-                throw java.lang.Exception("Empty response body")
+                throw HttpServiceException("Empty response body")
             }
         } else {
             val code = response.code
             val text = response.body?.string() ?: "unknown"
 
-            // TODO: Create custom exception
-            throw java.lang.Exception("Invalid response, code: $code, response: $text")
+            throw HttpServiceException("Invalid response, code: $code, response: $text")
```

Comment:
We should save original message and code in exception. Writing them in string requires users to parse it to get it.